src/lava/magma/core/model/py/ports.py

```diff
@@ -9,96 +9,229 @@
+
+    shape : tuple
+        The shape of the Port.
+
+    d_type: type
+        The data type of the Port.
+
+    Attributes
+    ----------
+    _csp_ports : list
+        A list of CSP Ports used by this IO Port.
+    """
+    def __init__(self,
+                 csp_ports: ty.List[AbstractCspPort],
+                 process_model: AbstractProcessModel,
+                 shape: ty.Tuple[int, ...],
+                 d_type: type):
+
+        self._csp_ports = csp_ports
+        super().__init__(process_model, shape, d_type)
+
+    @property
+    def csp_ports(self) -> ty.List[AbstractCspPort]:
+        """Property to get the corresponding CSP Ports of all connected
+        PyPorts (csp_ports). The CSP Port is the low level interface of the
+        backend messaging infrastructure which is used to send and receive data.
+
+        Returns
+        -------
+        A list of all CSP Ports connected to the PyPort.
+        """
+        return self._csp_ports
+
+
+class PyInPort(AbstractPyIOPort):
     """Python implementation of InPort used within AbstractPyProcessModel.
-    If buffer is empty, recv() will be blocking.
+
+    PyInPort is an input Port processing data sent from a connected output Port
+    (PyOutPort) over a channel. PyInPort can receive (recv()) the data, which
+    removes it from the channel, look at (peek()) at the data which keeps it on
+    the channel or check (probe()) if there is data on the channel.
+    The different class attributes are used to select the type of OutPorts via
+    LavaPyType declarations in PyProcModels, e.g., LavaPyType(
+    PyInPort.VEC_DENSE, np.int32, precision=24) creates a PyInPort.
+    A PyOutPort (source) can be connected to one or multiple PyInPorts (target).
+
+    Class attributes
+    ----------------
+    VEC_DENSE : PyInPortVectorDense, default=None
+        Type of PyInPort.
+
+    VEC_SPARSE : PyInPortVectorSparse, default=None
+        Type of PyInPort.
+
+    SCALAR_DENSE : PyInPortScalarDense, default=None
+        Type of PyInPort.
+
+    SCALAR_SPARSE : PyInPortScalarSparse, default=None
+        Type of PyInPort.
     """
 
     VEC_DENSE: ty.Type["PyInPortVectorDense"] = None
     VEC_SPARSE: ty.Type["PyInPortVectorSparse"] = None
     SCALAR_DENSE: ty.Type["PyInPortScalarDense"] = None
     SCALAR_SPARSE: ty.Type["PyInPortScalarSparse"] = None
 
-    def __init__(self, csp_recv_ports: ty.List[CspRecvPort], *args):
-        self._csp_recv_ports = csp_recv_ports
-        super().__init__(*args)
-
-    @property
-    def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
-        return self._csp_recv_ports
-
     @abstractmethod
     def recv(self):
+        """Abstract method to receive data (vectors/scalars) sent from connected
+        out Ports (source Ports). Removes the retrieved data from the channel.
+        Expects data on the channel and will block if there is no data to
+        retrieve on the channel.
+
```

Comment:
"will block the process from progressing to the next time step if there is no"...?

Reply:
Changed to "will block execution if there is no data to retrieve on the channel"